include/chaiscript/language/chaiscript_engine.hpp

```diff
@@ -213,11 +234,15 @@ namespace chaiscript
         throw chaiscript::exception::file_not_found_error(t_filename);
       }
 
-      const auto size = infile.tellg();
+      auto size = infile.tellg();
       infile.seekg(0, std::ios::beg);
 
       assert(size >= 0);
 
+      if (skip_bom(infile)) {
+          size-=3; // decrement the BOM size from file size, otherwise we'll get parsing errors
+      }
```

Comment:
I'd probably do another `asset(size >= 0);` here to verify there's actually something beyond the BOM.